src/applications/disability-benefits/686c-674/config/location-schema.js

```diff
@@ -0,0 +1,86 @@
+import { isChapterFieldRequired } from './helpers';
+
+export const locationUISchema = (
+  chapter,
+  outerField,
+  isInsideListLoop,
+  uiTitle,
+  formChapter,
+) => {
+  // IF we are inside a list loop, return hideIf and required uiSchema that use `index`
```

Comment:
Another option would be to always accept an `index` argument, and check to see if it's type is `number`. Then you could handle everything in one block and remove the if statement below.